addons/web/static/src/model/relational_model/dynamic_list.js

```diff
@@ -274,7 +275,10 @@ export class DynamicList extends DataPoint {
         if (!Object.keys(changes).length || record === this._recordToDiscard) {
             return;
         }
-        const validSelection = this.selection.filter((record) => {
+        const validSelection = this.selection.filter((record, index, selection) => {
+            if (!(selection.findIndex((rec) => rec.resId === record.resId) === index)) {
```

Comment:
It's potentially more costly to do that check rather than doing the modifier evaluation twice on the same record. I would ignore this case, unless you have a very good reason. I would thus call `unique` line 303 below.